### ISSUE TITLE
PR: octal and binary integer literals

### ISSUE THREAD

**Issue by Zekfad** [dart-lang/sdk#51634]
fixes https://github.com/dart-lang/language/issues/2708
fixes https://github.com/dart-lang/language/issues/581

Review on: https://dart-review.googlesource.com/c/sdk/+/286922

**Comment by lrhn** [comment]
Probably want to decide that we actually want them in the language before doing more about this. 
If we do, it needs an experiment flag (and it's probably not going into the 3.0 release, too close to add more features.)

And it will need many more tests.

**Comment by Zekfad** [comment]
You're totally right. That's prototype implementation, in case if there will be progress in future.

And, honestly, it was also interesting for me personally to see some internals of dart while trying to make this PR working.
